### PR TITLE
fix #57: multi value better handling

### LINKED ISSUE #57: Multiple and All values are send as 'value1 + value2 + …'
Hi,

When using multiple values or All options (with auto mode), the query sent to the Warp10 datasource is like that : 
`'value1 + value2 + value3 + …' variable STORE`

Warp10 didnt like it. A best way will be to send something like : 
`'~(value1|value2|value3|…)' variable STORE`

With All option, I can override with a custom value of '~.*' but it does not work with multiple option.

### PR DESCRIPTION
**Pull request by redref**
Working on #57 .
Tried to cover standard cases. Runned some tests, but they are not exhaustive.

### CODE REVIEW

**Comment by redref** [comment]
@tduboys : if you can try it and confirm, would be awesome !

**Comment by tduboys** [comment]
HI,
I confirm it works.
The patch creates a new variable called « <variable>_list » that contains all selected options on a list format
and a second variable <variable> with the REOPTALT option to be used on FETCH.

I only tested on a FETCH filter, but it's working on this case.
Thanks @redref

File: src/warp10-datasource.ts [hunk 1]
```diff
@@ -264,18 +264,36 @@ export default class Warp10Datasource {
...
-      if (myVar.current.value != null && (myVar.current.value === '$__all' || (myVar.current.value.length === 1 && myVar.current.value[0] === '$__all')))
-      {
-        if (myVar.allValue !== null)
-          value = myVar.allValue;
-        else
-          value = myVar.options.slice(1).map(e => e.text).join(" + ");
+      let value = myVar.current.text // single: "label", multi: "label1 + label2", all: "All"
+      let value_l = 'NULL';
+
+      if (myVar.current.value != null) {
```
Comment (miton18) [inline src/warp10-datasource.ts]: We can enhance it with:
```js
if (myVar.current.value == null) {
    continue
}
```

File: src/warp10-datasource.ts [hunk 2]
```diff
@@ -264,18 +264,36 @@ export default class Warp10Datasource {
...
+        // Apply "Custom all value" if applicable
+        if (myVar.current.value === '$__all' || (myVar.current.value.length === 1 && myVar.current.value[0] === '$__all')) {
```
Comment (miton18) [inline src/warp10-datasource.ts]: I think `myVar.current.value.length === 1` can have a weird behaviour depending of current value type (string or array)

File: src/warp10-datasource.ts [hunk 3]
```diff
@@ -264,18 +264,36 @@ export default class Warp10Datasource {
...
+          value_l = "[ ".concat(myVar.options.slice(1).map(e => "'" + e.value + "'").join(" "), " ]");
```
Comment (miton18) [inline src/warp10-datasource.ts]:
```js
 value_l = `[ ${ myVar.options.slice(1).map(e => `'${ e.value }'`).join(" ") } ]`;
```

File: src/warp10-datasource.ts [hunk 4]
```diff
@@ -264,18 +264,36 @@ export default class Warp10Datasource {
...
+          if (myVar.allValue !== null && myVar.allValue != "") {
+            value = "'" + myVar.allValue + "'";
```
Comment (miton18) [inline src/warp10-datasource.ts]: value = `'${ myVar.allValue }'`;

**Review by miton18 — CHANGES_REQUESTED** [review]

File: src/warp10-datasource.ts (hunk 1 above)
Comment (redref) [reply src/warp10-datasource.ts]: This is a behavior change.
Nevertheless, I think we can do it safely as grafana will provide value no matter what.

**Comment by pi-r-p** [comment]
@redref : Sorry, I worked on the same subject yesterday night and tonight... And I just discovered you also did a PR for the same issue. 
We may discuss it on monday maybe ?

**Comment by redref** [comment]
@pi-r-p : Closing this, despite my concerns on your PR, it is better styled and cover a bit more.

**Pull request closed without merging**